x/precisebank/types/fractional_balances_test.go

```diff
@@ -84,3 +85,53 @@ func TestFractionalBalances_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestFractionalBalances_SumAmount(t *testing.T) {
+	generateRandomFractionalBalances := func(n int) (types.FractionalBalances, sdkmath.Int) {
+		balances := make(types.FractionalBalances, n)
+		sum := sdkmath.ZeroInt()
+
+		for i := 0; i < n; i++ {
+			addr := sdk.AccAddress{byte(i)}.String()
+			amount := sdkmath.NewInt(rand.Int63())
+			balances[i] = types.NewFractionalBalance(addr, amount)
+
+			sum = sum.Add(amount)
+		}
+
+		return balances, sum
+	}
+
+	multiBalances, sum := generateRandomFractionalBalances(10)
```

Comment:
can you add a test that checks the behavior for a bonkers number of accounts?
like what happens in the unlikely scenario in which there are enough partial balances to overflow sdkmath.Int?

maybe not worth adding the test case? lol. but maybe we do just to show we thought about it.

as an aside... the max fractional balance for a single account is (1e12 - 1), so it would take more than $10^{65}$ accounts to actually make an overflow happen. 😂 not to mention the fact that a total supply of ~1e9 KAVA means we can't overflow Int on mainnet if we tried. only 1.08e27 akava even exists. we'd need a total supply of 1e47 KAVA to overflow on conversion to akava

Reply:
just added a basic test (with an invalid FractionalBalance max int) at least to verify it will panic instead of some overflow